### PR TITLE
Chef 17 compatiblity changes

### PR DESCRIPTION
**Pull request by bmhughes**
# Description

Chef 17 compatiblity changes

NOTE: Until chef/chef#11497 is merged, the systemd suites will fail.

## Issues Resolved

- n/a

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 driver:
   name: dokken
   privileged: true
-  chef_version: <%= ENV['CHEF_VERSION'] || 'latest' %>
+  chef_version: <%= ENV['CHEF_VERSION'] || '17.1.16' %>
```
Comment (Xorima) [inline kitchen.dokken.yml:5]: can we reset this to latest before merging please

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@bmhughes I can't replicate the testing failures on the two platforms in the `delete` suite. For me the only failure is the process check (Mine doesn't show `Ssl`, it shows `Ss` instead). Any ideas why this is happening on the CI?

**Comment by bmhughes** [comment]
It seems past-me ran into this before as Ubuntu 1804/2004 was disabled for the delete suite in the GH CI config, and I guess this was why.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,8 +52,6 @@ jobs:
           - 'default'
           - 'delete'
         exclude:
-          - os: 'centos-6'
-            suite: 'delete'
           - os: 'ubuntu-1804'
```
Comment (ramereth) [inline .github/workflows/ci.yml:55]: Might be worth adding a comment as to why these are excluded so we don't go through this again.

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.0.0](https://github.com/sous-chefs/dhcp/releases/tag/8.0.0)

**Comment by damacus** [comment]
Resolves #172

**Pull request merged**